### PR TITLE
gcc fixes for #672946 and #677052

### PR DESCRIPTION
**Pull request by stefson**
Firefox can be build with lto, but the binary segfaults for some users. This fixes one of the possible underlying problems. 

Also fixes compile for qtgui-5.12

### CODE REVIEW

File: sys-devel/gcc/gcc-8.3.0.ebuild [hunk 1]
```diff
@@ -21,6 +21,10 @@ fi
 src_prepare() {
 	toolchain_src_prepare
 
+	# https://gcc.gnu.org/bugzilla/show_bug.cgi?id=69471
+	# https://bugs.gentoo.org/672946
+	epatch "${FILESDIR}"/gcc-8.3.0-_fix_avx2_pr69471.patch
```
Comment (blueness) [inline sys-devel/gcc/gcc-8.3.0.ebuild]: this needs to be within the if statement that checks for the elibc_musl.

File: sys-devel/gcc/files/gcc-8.3.0-_fix_avx2_pr69471.patch [hunk 2]
```diff
@@ -0,0 +1,49 @@
+--- a/gcc/config/i386/i386.opt	2019/02/23 18:44:54	269163
++++ b/gcc/config/i386/i386.opt	2019/02/23 19:39:35	269164
```
Comment (blueness) [inline sys-devel/gcc/files/gcc-8.3.0-_fix_avx2_pr69471.patch]: There's no way I can add a patch to an important component like gcc unless I know where its coming from.  If its coming from upstream gcc, then they should not go here but into the patchset maintained by slyfox.

Comment (stefson) [reply sys-devel/gcc/files/gcc-8.3.0-_fix_avx2_pr69471.patch]: ok, so lets wait for patchset revbump.

File: sys-devel/gcc/gcc-8.3.0.ebuild (hunk 1 above)
Comment (stefson) [reply sys-devel/gcc/gcc-8.3.0.ebuild]: I read this as 
		`cpu_indicator.patch`
		`posix_memalign.patch`
to be applied if building a cross toolchain with musl as a libc.

**Comment by stefson** [comment]
Ok, I'll wait for a revbump of the patchset and am going to overwrite this pullrequest with it. Feel free to remove the outdated -r5 ebuild.

File: sys-devel/gcc/gcc-8.3.0.ebuild (hunk 1 above)
Comment (blueness) [reply sys-devel/gcc/gcc-8.3.0.ebuild]: No, || = or.  So either  elibc_musl *or* [[ ${CATEGORY} = cross-*-musl* ]].

Comment (stefson) [reply sys-devel/gcc/gcc-8.3.0.ebuild]: Well, if this is what you want you should have a look at the gcc-7 ebuilds. All of them have an extra patch outside of that statement, and it's where I've been looking for an example of common practice.

**Comment by stefson** [comment]
@blueness - the fix got merged into gcc-8.3.0, therefore I removed the patch and added gcc-7.4.0-r1 to the pullrequest.

**Comment by blueness** [comment]
@stefson why do we need this in 7.4.0-r1?  can't it be merged upstream for that version of gcc too?  i'm trying to empty out the musl overlay, not become more dependent on it.  if the patch was accepted for 8.3.0 then its backport should be acceptible for 7.4.0.  else just use 8.3.0.

**Comment by stefson** [comment]
I don't know why this gets pulled in. It should be a removal of 8.2.0-r5 and sync of gcc-7.4.0-r1 with the tree.

**Comment by blueness** [comment]
is the overlay out of sync with the tree?

**Pull request closed without merging**